app/views/samples_reports/show.haml

```diff
@@ -331,5 +330,62 @@
     }
   }
 
+  function getSvg(chartId, chartName){
+    var svgData = document.getElementById(chartId);
+    applyInline(svgData);
+    svgData = svgData.outerHTML;
+    return svgData;
+  }
+
+  async function printPDF() {
+    var context = getUrlParameter('context');
+    
+    var measured_signal_svg = encodeURIComponent(getSvg("barchart", "measured_signal"));
+    var specific_svg;
+    if ("#{@purpose}" == "Challenge")
+      specific_svg = encodeURIComponent(getSvg("rocchart", "roc_curve"));
+    else
+      specific_svg = encodeURIComponent(getSvg("linechart", "limit_of_detection"));
+
+    // if purpose is challenge, then we need the live threshold calculations
+    if (#{@purpose.to_json} == "Challenge") {
+      var threshold = document.getElementById("threshold").value;
+      var url = '/samples_reports/update_threshold?context='+context+"&samples_report_id=#{@samples_report.id}&threshold="+threshold;
```

Comment:
Why do you need to make the calculation on the backend now, just before sending the charts to the backend for rendering? Didn't you already make that request to live update the charts?